custom/icds/utils/custom_sms_report.py

```diff
@@ -0,0 +1,43 @@
+import json
+from dimagi.utils.couch.cache.cache_core import get_redis_client
+
+
+class CustomSMSReportTracker(object):
+    def __init__(self):
+        self.client = get_redis_client()
+
+    @property
+    def report_key(self):
+        return 'custom-sms-report-tracker'
+
+    @property
+    def active_reports(self):
+        unfiltered_report_list = self.client.get(self.report_key)
+        if(unfiltered_report_list):
+            reports = json.loads(unfiltered_report_list)
+        else:
+            reports = []
+        return reports
+
+    def clean_all_active_reports(self):
+        self.client.delete(self.report_key)
+
+    def remove_report_from_list(self, start_date, end_date):
+        reports = self.active_reports
+        updated_reports = [report for report in reports
+                    if report['start_date'] != start_date or report['end_date'] != end_date]
```

Comment:
nit: `retain_reports`

Depeding on other usages, 
can we use a simple string (`f"{start_date}-{end_date}"`) instead of a dict?
then this will become simply, `reports.remove(f"{start_date}-{end_date}")`

Reply:
if possible we should also make this atomic, i think thats a bit challenging with redis.

Reply:
> can we use a simple string (f"{start_date}-{end_date}") instead of a dict?

I was also thinking to do that but I thought if we need to save some extra info then it would be some work later. So using a dict gives the flexibility of having support for some custom fields in the future. What are your thoughts on it? @mkangia

> if possible we should also make this atomic, i think thats a bit challenging with Redis.

It is a nice suggestion and maybe it can cause issues if two reports finish at the same time. But given the use-case, should we actually do it? If yes, can you point to some examples of how it is being used in HQ.

Reply:
> we need to save some extra info then it would be some work later

I think it will be okay to put that later and keep things simple for now, until we really need a dictionary/more info.

Reply:
> But given the use-case, should we actually do it? 

You saying this not the use-case for keeping it atomic? For an example, I think `MessagingRuleProgressHelper.mark_shard_complete` does something like that.

Reply:
>I think it will be okay to put that later and keep things simple for now, until we really need a dictionary/more info.

Cool. Will remove dict.

> You saying this not the use-case for keeping it atomic?

This is definitely something race condition can arise, I was just thinking that it's use would not be that frequent, so do we need to take that extra step of making it atomic? 

> I think MessagingRuleProgressHelper.mark_shard_complete does something like that

Thanks. I will look into it.

Reply:
> This is definitely something race condition can arise, I was just thinking that it's use would not be that frequent, so do we need to take that extra step of making it atomic

Agreed it wont be used a lot but definitely worth adding. Feel free to take it as a followup and not a part of this PR itself if it looks like something you dont want to block on.

Reply:
I am leaving it here now. Will look into making it atomic later.